lib/java/spring/events/src/main/java/co/airy/spring/events/filters/HttpLogFilter.java

```diff
@@ -0,0 +1,76 @@
+package co.airy.spring.events.filters;
+
+import co.airy.spring.events.custom.HttpEventPublisher;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.server.ServletServerHttpRequest;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+import org.springframework.web.util.ContentCachingRequestWrapper;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@Component
+public class HttpLogFilter extends OncePerRequestFilter {
+
+    @Autowired
+    HttpEventPublisher httpEventPublisher;
+
+    private static final int MAX_JSON_LENGTH = 2048;
+
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain) throws ServletException, IOException {
+        boolean isFirstRequest = !this.isAsyncDispatch(request);
+        HttpServletRequest requestToUse = request;
+        if (isFirstRequest && !(request instanceof ContentCachingRequestWrapper)) {
+            requestToUse = new ContentCachingRequestWrapper(request, MAX_JSON_LENGTH);
+        }
+
+        try {
+            filterChain.doFilter(requestToUse, response);
+        } finally {
+            publishEvent(requestToUse);
```

Comment:
I've just realized this is the opposite of what we talked about. It will log the request only after processing it